indexing-service/src/main/java/org/apache/druid/indexing/common/tasklogs/ConsoleLoggingEnforcementConfigurationFactory.java

```diff
@@ -117,16 +123,20 @@ private Appender findConsoleAppender()
     }
 
     /**
-     * remove all appenders from a logger and append a console appender to it
+     * Ensure there is a console logger defined. Without a console logger peon logs wont be able to be stored in deep storage
      */
     private void applyConsoleAppender(LoggerConfig logger, Appender consoleAppender)
     {
-      if (logger.getAppenderRefs().size() == 1
-          && logger.getAppenderRefs().get(0).getRef().equals(consoleAppender.getName())) {
-        // this logger has only one appender and its the console appender
-        return;
+      // Alter log level for this class to be warning. This needs to happen because the logger is using the default
+      // config, which is level error and appends to console, since the logger is being configured here.
+      Configurator.setLevel(LogManager.getLogger(ConsoleLoggingEnforcementConfigurationFactory.class).getName(), Level.WARN);
```

Comment:
Should this be done once only instead of on each call to `applyConsoleAppender`?

Also do not call `LogManager.getLogger(ConsoleLoggingEnforcementConfigurationFactory.class)` in here. It is slow. Can we reference the static logger instead?

Reply:
done. needed to alter the druid logger class to make getName available. testing out the change to make sure it still works as expected